### PR TITLE
Protobuf changes for protobuf.js instances

### PR DESCRIPTION
**Pull request by twojtasz**
Protobuf.js uses prototype object creation which has the effect of
looking up the prototype chain for properties even if the instance
does not have it set.
    
This requires the use of 'hasOwnProperty()' in some places to
distinguish between the instance and the prototype property.
    
The library also assigns default values for fields, such as the
empty array or a value of 0. Our parsing would look at presences
to determine if a property was meaningfully set, but now we need to
have additional validation due to the default values set by the
protobuf.js library.

### CODE REVIEW

File: modules/io/src/common/loaders.js [hunk 1]
```diff
@@ -198,6 +199,78 @@ function getPBEXVIZType(arrayBuffer) {
...
+ *
+ * The following mutations are made:
+ *
+ * - Remove empty Arrays from the instance, protobuf.js will set to the default value: [] (the empty array)
+ * - Enum => String
+ * - Handle arrays of enum
+ * - Recursive on objects
+ */
+/* eslint-disable max-depth, complexity */
+function postProcessProtobuf(msg, pbType) {
```
Comment (jlisee) [inline modules/io/src/common/loaders.js]: I think deserves a dedicated test function really.  With my experience in Protobuf.js with this you can easily miss some corner cases and a test makes it much easier to absorb new things.

Comment (twojtasz) [reply modules/io/src/common/loaders.js]: Yes, not that I have fixed the ui_config structure handling i'll will beef up code coverage

**Comment by twojtasz** [comment]
@jlisee 
> * How fast is postProcessProtobuf vs. toObject?
The postProcessProtobuf on average negligible compared to the regular `decode(msg)`.  Where as to `toObject()`  makes it jump from ~5ms to ~30ms

> * Do we have good test coverage for this?
I will be looking to verify our coverage for this flow.  I know the round-trip in @xviz/schema is all passing, but that does not cover this. I"ll be adding cases using the @xviz/builder with the ProtobufWriter next
